File: src/cleansia_ios/CleansiaCore/Sources/CleansiaCore/Config/CleansiaWeb.swift
import Foundation

/// The public web app both mobile apps link out to. A move off `.cz` (a `.eu`
/// domain is under consideration) must be a one-line edit here — never a grep
/// across two apps and five locale catalogs, so no other Swift file, string
/// catalog or plist may spell the domain out.
public enum CleansiaWeb {
    public static let domain = "cleansia.cz"

    public static let origin = "https://\(domain)"

    public static let supportEmail = "support@\(domain)"

    /// Routed by the customer web app (`app.routes.ts`).
    public static var termsURL: URL {
        url("/terms")
    }

    public static var privacyURL: URL {
        url("/privacy")
    }

    public static func referralLink(code: String) -> String {
        "\(origin)/r/\(code)"
    }

    private static func url(_ path: String) -> URL {
        guard let url = URL(string: origin + path) else {
            fatalError("CleansiaWeb.origin is malformed: \(origin)")
        }
        return url
    }
}

File: src/cleansia_android/core/src/main/java/cz/cleansia/core/config/CleansiaWeb.kt
package cz.cleansia.core.config

/**
 * The public web app both Android apps link out to. Mirrors
 * `CleansiaCore/Sources/CleansiaCore/Config/CleansiaWeb.swift` on iOS: a move
 * off `.cz` (a `.eu` domain is under consideration) must be a one-line edit
 * here — never a grep across two apps and ten `strings.xml` files. That is why
 * the consent sentences carry [ConsentLink] placeholders instead of real
 * addresses, and why `ConsentCatalogTest` fails any translation that spells
 * [DOMAIN] out.
 */
object CleansiaWeb {
    const val DOMAIN = "cleansia.cz"

    const val ORIGIN = "https://$DOMAIN"

    /** Routed by the customer web app (`app.routes.ts`). */
    const val TERMS_URL = "$ORIGIN/terms"

    const val PRIVACY_URL = "$ORIGIN/privacy"
}
